common/src/main/java/org/valkyrienskies/mod/mixin/mod_compat/vanilla_renderer/MixinLevelRendererVanilla.java

```diff
@@ -178,12 +179,25 @@ private void redirectRenderChunkLayer(final LevelRenderer receiver,
                 center.x(), center.y(), center.z(),
                 camX, camY, camZ);
 
+            final AtomicReference<Matrix4f> matrix4fNew = new AtomicReference<>(matrix4f);
```

Comment:
Why are these atomic? Atomic is generally slow so unless this needs to be run on multiple threads we should use something else.

Reply:
Just write something like `class MutableObject<T> { var object: T }`. Its effectively a modifiable reference but it won't cause lag.